### PR TITLE
Yanan ide 4656

### PR DESCRIPTION
**Pull request by simonjhy**

### CODE REVIEW

**Comment by jtydhr88** [comment]
SF Successfully on liferay-ide folder

File: tools/plugins/com.liferay.ide.server.ui/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -38,7 +38,8 @@ Require-Bundle: com.liferay.ide.core,
  org.eclipse.wst.sse.ui,
  org.eclipse.wst.ws,
  org.eclipse.wst.ws.explorer,
- org.eclipse.wst.ws.parser
+ org.eclipse.wst.ws.parser,
+ org.apache.commons.lang3
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.server.ui/META-INF/MANIFEST.MF]: the list should be ordered

File: tools/plugins/com.liferay.ide.server.core/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -24,7 +24,8 @@ Require-Bundle: com.liferay.ide.core,
  org.eclipse.wst.common.project.facet.core,
  org.eclipse.wst.server.core,
  org.eclipse.wst.validation,
- org.json
+ org.json,
+ com.google.guava
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.server.core/META-INF/MANIFEST.MF]: the list should be ordered

File: tools/plugins/com.liferay.ide.server.ui/META-INF/MANIFEST.MF (hunk 1 above)
Comment (simonjhy) [reply tools/plugins/com.liferay.ide.server.ui/META-INF/MANIFEST.MF]: fixed

File: tools/plugins/com.liferay.ide.server.core/META-INF/MANIFEST.MF (hunk 2 above)
Comment (simonjhy) [reply tools/plugins/com.liferay.ide.server.core/META-INF/MANIFEST.MF]: fixed

**Comment by jtydhr88** [comment]
SF Successfully on liferay-ide folder

File: tools/plugins/com.liferay.ide.server.ui/src/com/liferay/ide/server/ui/editor/PortalServerLaunchEditorSection.java [hunk 3]
```diff
@@ -345,6 +341,21 @@ protected void validate() {
 				}
 			}
 
+			String[] memoryArrays = StringUtils.split(memoryArgs.getText(), " ");
+
+			boolean memoryArgInvalid = Stream.of(
+				memoryArrays
+			).filter(
+				arg -> !arg.startsWith("-Xm")
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.server.ui/src/com/liferay/ide/server/ui/editor/PortalServerLaunchEditorSection.java]: we should give more details on the error message, it is hard to see the rule from error message

File: tools/plugins/com.liferay.ide.server.ui/src/com/liferay/ide/server/ui/editor/PortalServerLaunchEditorSection.properties [hunk 4]
```diff
@@ -5,4 +5,5 @@ externalPropertiesLabel=External properties:
 invalidExternalPropertiesFile=Invalid external properties file
 liferayLaunch=Liferay Launch
 memoryArgsLabel=Memory args:
-useDeveloperMode=Use developer mode
\ No newline at end of file
+useDeveloperMode=Use developer mode
+memoryArgumentsInvalid=Memory arguments setting is invalid.
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.server.ui/src/com/liferay/ide/server/ui/editor/PortalServerLaunchEditorSection.properties]: we should give more details on the error message, it is hard to see the rule from error message

**Pull request closed without merging**